src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -93,55 +229,64 @@
-            final String[] tokens = rawArgumentValue.split(FEATURE_ARGUMENT_TAG_DELIMITER, -1);
-            final String usage = "Argument must either be a file, or of the form logical_name:file or logical_name(,key=value)*:feature_file";
+            //Use negative look ahead to avoid splitting URIs into multiple tokens
+            //i.e. someName:file://somefile -> ["someName", "file://somefile"]
+            final String MATCH_NAME_BUT_NOT_URI = "(?!" + URI_SCHEME_SEPARATOR + ")[" + FEATURE_ARGUMENT_TAG_DELIMITER + "]";
+            final String[] tokens = rawArgumentValue.split(MATCH_NAME_BUT_NOT_URI, -1);
 
-            // Check for malformed argument values
-            if (tokens.length > 2 || tokens.length == 0) {
-                throw new UserException.BadArgumentValue("", rawArgumentValue, usage);
-            }
-            for (final String token : tokens) {
-                if (token.isEmpty()) {
-                    throw new UserException.BadArgumentValue("", rawArgumentValue, "Empty name/file encountered. " + usage);
-                }
+            if( Arrays.stream(tokens).anyMatch(String::isEmpty)){
+                    throw new UserException.BadArgumentValue("", rawArgumentValue, "Empty name/file encountered. " + USAGE);
             }
 
-            if (tokens.length == 1) {
+            if (tokens.length == 0 || tokens.length > 2) {
+                throw new UserException.BadArgumentValue("", rawArgumentValue, USAGE);
+            } else if (tokens.length == 1) {
                 // No user-specified logical name for this FeatureInput, so use the absolute path to the File as its name
-                final File featureFile = new File(tokens[0]);
-                return new ParsedArgument(featureFile.getAbsolutePath(), featureFile);
+                final String featurePath = tokens[0];
+                return new ParsedArgument(FeatureInput.makeIntoAbsolutePath(featurePath), featurePath);
+            } else {
+                final String[] tags = splitTags(tokens[0], rawArgumentValue);
+                final String path = tokens[1];
+                return getNamedArgument(tags, path, rawArgumentValue);
             }
+        }
 
-            // User specified a logical name (and optional list of key-value pairs)
-            // for this FeatureInput using name(,key=value)*:File syntax.
-            // eg foo:file.vcf
-            // eg foo,a=3,b=false,c=fred:file.vcf
-            final String[] subtokens= tokens[0].split(FEATURE_ARGUMENT_KEY_VALUE_PAIR_DELIMITER, -1);
-            if (subtokens[0].isEmpty()){
-                throw new UserException.BadArgumentValue("", rawArgumentValue, usage);
-            }
-            final ParsedArgument pa= new ParsedArgument(subtokens[0], new File(tokens[1]));
+        private static ParsedArgument getNamedArgument(String[] tags, String path, String rawArgumentValue) {
+            final ParsedArgument pa= new ParsedArgument(tags[0], path);
```

Comment:
Nevermind, I see it below. I'm not sure that splitting this code up has made it more readable.